scipy/stats/_ksstats.py

```diff
@@ -0,0 +1,607 @@
+    k2b = (2 * zfour - 5 * zsquared) * pisquared / 4
+    k2c = pifour * (1 - 2 * zsquared) / 16
+
+    k3d = pisix * (5 - 30 * zsquared) / 64
+    k3c = pifour * (-60 * zsquared + 212 * zfour) / 16
+    k3b = pisquared * (135 * zfour - 96 * zsix) / 4
+    k3a = -30 * zsix - 90 * z**8
+
+    K0to3 = np.zeros(4)
+    # Use a Horner scheme to evaluate sum c_i q^(i^2)
+    # Reduces to a sum over odd integers.
+    maxk = int(np.ceil(16 * z / np.pi))
+    for k in range(maxk, 0, -1):
+        m = 2 * k - 1
+        msquared = m ** 2
+        mfour = m ** 4
+        msix = m ** 6
+        qpower = np.power(q, 8 * k)
+        coeffs = np.array([1.0,
+                           k1a + k1b * msquared,
+                           k2a + k2b*msquared + k2c * mfour,
+                           k3a + k3b*msquared + k3c * mfour + k3d * msix])
+        K0to3 *= qpower
+        K0to3 += coeffs
+    K0to3 *= q
+    K0to3 *= _SQRT2PI
+    # z**10 > 0 as z > 0.04
+    K0to3 /= np.array([z, 6 * zfour, 72 * z**7, 6480 * z**10])
+
+    # Now do the other sum over the other terms, all integers k
+    q = np.exp(-pisquared / 2 / zsquared)
+    # Coefficients of terms in the sums for K3
+    k2b = np.pi**2
+    k3b = 3 * pisquared * zsquared
+    k3c = -pifour
+    K0to3Extra = np.zeros(2)
+
+    for k in range(maxk, 0, -1):
+        ksquared = k**2
+        kfour = k**4
+        qpower = np.power(q, 2 * k + 1)
+        coeffs = np.array([k2b * ksquared, k3b * ksquared + k3c * kfour])
+        K0to3Extra *= qpower
+        K0to3Extra += coeffs
+    K0to3Extra *= q
+
+    zthree = z**3
+    K0to3Extra *= _SQRT2PI
+    K0to3Extra /= np.array([-36 * zthree, 216 * zsix])
+
+    K0to3[2:4] += K0to3Extra
+    powers_of_n = np.power(n * 1.0, np.arange(0, len(K0to3)) / 2.0)
+    K0to3 /= powers_of_n
+
+    if not cdf:
+        K0to3 *= -1
+        K0to3[0] += 1
+
+    Ksum = sum(K0to3)
+    return Ksum
+
+
+def _kolmogn(n, x, cdf=True):
+    """Compute the CDF(SF) for the two-sided Kolmogorov-Smirnov statistic.
+
+    x must be of type float, n of type integer.
+
+    Simard & L'Ecuyer (2011) [7]."""
+    if int(n) != n or n <= 0:
+        return np.nan
+    if x >= 1.0:
+        return _select_and_pin_prob(1.0, 0.0, cdf=cdf)
+    if x <= 0.0:
+        return _select_and_pin_prob(0.0, 1.0, cdf=cdf)
+    t = n * x
+    if t <= 1.0:  # Ruben-Gambino
+        if t <= 0.5:
+            return _select_and_pin_prob(0.0, 1.0, cdf=cdf)
+        prd = 1.0
+        mlt = 2 * t - 1
+        for m in range(1, n + 1):
+            prd = prd * m * mlt / n
+        return _select_and_pin_prob(prd, 1.0 - prd, cdf=cdf)
+    if t >= n - 1:  # Ruben-Gambino
+        onemx = 1.0 - x
+        prob = 2 * onemx**n
+        return _select_and_pin_prob(1 - prob, prob, cdf=cdf)
+    if x >= 0.5:  # Exact: 2 * smirnov
+        prob = 2 * scipy.special.smirnov(n, x)
+        return _select_and_pin_prob(1.0 - prob, prob, cdf=cdf)
+
+    nxsquared = t * x
+    if n <= 140:
+        if nxsquared <= 0.754693:
```

Comment:
These limits (0.754693, 4.0) came from section 4(i). Can you point me to the section where you see nxsquared <= 0.25 and n <= 140? I think that Ruben-Gambino has already been applied where it can be by the time the code gets here.

Reply:
You are righr, RG has already been applied.